src/platform/renesas/SystemTimeSupport.cpp

```diff
@@ -0,0 +1,172 @@
+#endif
+} // unnamed namespace
+
+/**
+ * Returns the number of FreeRTOS ticks since the system booted.
+ *
+ * NOTE: The default implementation of this function uses FreeRTOS's
+ * vTaskSetTimeOutState() function to get the total number of ticks,
+ * irrespective of tick counter overflows.  Unfortunately, this function cannot
+ * be called in interrupt context, no equivalent ISR function exists, and
+ * FreeRTOS provides no portable way of determining whether a function is being
+ * called in an interrupt context.  Adaptations that need to use the Chip
+ * Get/SetClock methods from within an interrupt handler must override this
+ * function with a suitable alternative that works on the target platform.  The
+ * provided version is safe to call on ARM Cortex platforms with CMSIS
+ * libraries.
+ */
+
+uint64_t FreeRTOSTicksSinceBoot(void) __attribute__((weak));
+
+uint64_t FreeRTOSTicksSinceBoot(void)
+{
+    TimeOut_t timeOut;
+
+#ifdef __CORTEX_M
+    if (SCB->ICSR & SCB_ICSR_VECTACTIVE_Msk) // running in an interrupt context
+    {
+        // Note that sNumOverflows may be quite stale, and under those
+        // circumstances, the function may violate monotonicity guarantees
+        timeOut.xTimeOnEntering = xTaskGetTickCountFromISR();
+        timeOut.xOverflowCount  = sNumOfOverflows;
+    }
+    else
+    {
+#endif
+
+        vTaskSetTimeOutState(&timeOut);
+
+#ifdef __CORTEX_M
+        // BaseType_t is supposed to be atomic
+        sNumOfOverflows = timeOut.xOverflowCount;
+    }
+#endif
+
+    return static_cast<uint64_t>(timeOut.xTimeOnEntering) + (static_cast<uint64_t>(timeOut.xOverflowCount) << kTicksOverflowShift);
+}
+
+Clock::Microseconds64 ClockImpl::GetMonotonicMicroseconds64(void)
+{
+    return Clock::Microseconds64((FreeRTOSTicksSinceBoot() * kMicrosecondsPerSecond) / configTICK_RATE_HZ);
+}
+
+Clock::Milliseconds64 ClockImpl::GetMonotonicMilliseconds64(void)
+{
+    return Clock::Milliseconds64((FreeRTOSTicksSinceBoot() * kMillisecondsPerSecond) / configTICK_RATE_HZ);
+}
+
+uint64_t GetClock_Monotonic(void)
+{
+    return (FreeRTOSTicksSinceBoot() * kMicrosecondsPerSecond) / configTICK_RATE_HZ;
+}
+
+uint64_t GetClock_MonotonicMS(void)
+{
+    return (FreeRTOSTicksSinceBoot() * kMillisecondsPerSecond) / configTICK_RATE_HZ;
+}
+
+uint64_t GetClock_MonotonicHiRes(void)
+{
+    return GetClock_Monotonic();
+}
+
+CHIP_ERROR ClockImpl::GetClock_RealTime(Clock::Microseconds64 & aCurTime)
+{
+    // TODO(19081): This platform does not properly error out if wall clock has
+    //              not been set.  For now, short circuit this.
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+#if 0
+    if (sBootTimeUS == 0)
+    {
+        return CHIP_ERROR_REAL_TIME_NOT_SYNCED;
+    }
+    aCurTime = Clock::Microseconds64(sBootTimeUS + GetClock_Monotonic());
+    return CHIP_NO_ERROR;
+#endif
+}
+
+CHIP_ERROR ClockImpl::GetClock_RealTimeMS(Clock::Milliseconds64 & aCurTime)
+{
+    struct m_timeval tv;
+    clock_gettime(&tv);
```

Comment:
this function could return an error (not 0) Can it be handled.

Reply:
Handled the error. Returned timenotsynced, because I probably don't know exactly what went wrong (returns -1)